src/hooks/useScrollWatcher.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+import throttle from 'lodash.throttle';
+
+function useScrollWatcher(ref, cb, throttlingTimeout = 0) {
+  useEffect(() => {
+    if (!cb) {
+      return;
+    }
+    const scrollingElement = ref.current;
+    const onScroll = throttle(() => {
+      cb(scrollingElement.scrollTop);
+    }, throttlingTimeout);
+
+    scrollingElement.addEventListener('scroll', onScroll);
+
+    return () => {
+      cb(0);
```

Comment:
Should it be safer to call `cb(0)` after having removed the listener ? I'm afraid there is a probability where a `onScroll` event could be fired between this line and next one.